datafusion/expr/src/function.rs

```diff
@@ -376,6 +381,9 @@ pub fn signature(fun: &BuiltinScalarFunction) -> Signature {
         BuiltinScalarFunction::Chr | BuiltinScalarFunction::ToHex => {
             Signature::uniform(1, vec![DataType::Int64], fun.volatility())
         }
+        BuiltinScalarFunction::Greatest | BuiltinScalarFunction::Least => {
+            Signature::variadic_equal(fun.volatility())
```

Comment:
Should be `Signature::variadic_equal(SUPPORTED_COMPARISON_TYPES, fun.volatility())`